src/main/java/net/mcreator/workspace/elements/ModElement.java

```diff
@@ -317,6 +317,10 @@ public void setRegistryName(String registry_name) {
 		return path;
 	}
 
+	public void setFolderPath(String path) {

```

Comment:
Use setParentFolder as it it made for this purpose, and after renaming if you do it properly you should have new FolderElement anyways

Reply:
Still, i don't find how to do it

Reply:
Set name of ((FolderElement) selected).setName(newName); before calling setParentFolder and then set parent folder to this same folder to update the path.

Reply:
you mean like this?
`			((FolderElement) selected).setName(newName);
			for (ModElement modElement : mcreator.getWorkspace().getModElements()) {
				if (currentFolder.equals(modElement.getFolderPath())) {
					modElement.setParentFolder(((FolderElement) selected));
				}
			}
			for (FolderElement childFolder : currentFolder.getRecursiveFolderChildren()) {
				for (ModElement modElement : mcreator.getWorkspace().getModElements()) {
					if (childFolder.equals(modElement.getFolderPath())) {
						modElement.setParentFolder(((FolderElement) selected));
					}
				}
			};`

Reply:
you mean like this?
		
![image](https://user-images.githubusercontent.com/60476871/120220034-2ee59400-c24d-11eb-8518-e34eb137ab94.png)

Reply:
(probably not this as it breaks things up)